JavaScript/chapter01/1.5 - OneAway/rroque98_sol.js

```diff
@@ -4,40 +4,65 @@ Given 2 strings write a function that will check if it is
 1 or 0 edits away*/
 
 const isOneAway = (str1, str2) => {
+  const diffInLen = Math.abs(str1.length - str2.length);
+  if (diffInLen > 1) {
+    return false;
+  }
+
+  let errorCount = 0;
   if (str1.length === str2.length) {
-    let errorCount = 0;
     for (let i = 0; i < str1.length; i++) {
```

Comment:
it might be better to use the [guard clause pattern](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html):

```javascript
if (str1.length === str2.length) {
  // ..
  return true;
}
// rest of the code.
```